### PR TITLE
Add OCaml 5.3 previews for merlin, merlin-lib and ocaml-lsp-server

### PR DESCRIPTION
**Pull request by kit-ty-kate**

### CODE REVIEW

File: packages/dot-merlin-reader/dot-merlin-reader.5.3~5.3preview/opam [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+license:      "MIT"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "5.3" & < "5.4"}
+  "dune" {>= "2.9.0"}
+  "merlin-lib" {= version}
+  "ocamlfind" {>= "1.6.0"}
```
Comment (voodoos) [inline packages/dot-merlin-reader/dot-merlin-reader.5.3~5.3preview/opam:17]:
```suggestion
  "ocamlfind" {>= "1.6.0"}
  "alcotest" {with-test}
```

File: packages/merlin/merlin.5.3~5.3preview/opam [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+available: opam-version >= "2.1.0"
+depends: [
+  "dune" {>= "3.0.0"}
+  "ocaml" {>= "5.3" & < "5.4"}
+  "merlin-lib" {= version}
+  "dot-merlin-reader" {= version}
+  "ocaml-index" {>= "1.0" & post}
+  "yojson" {>= "2.0.0"}
+  "conf-jq" {with-test}
+  "ppxlib" {with-test}
```
Comment (voodoos) [inline packages/merlin/merlin.5.3~5.3preview/opam:23]:
```suggestion
  "ppxlib" {with-test}
  "alcotest" {with-test}
```

**Review by voodoos — COMMENTED** [review]
I forgot about this: we should attach the alcotest-run tests to the merlin-lib package and not the others. Meanwhile this should fix one of the CI issues.

File: packages/ocaml-lsp-server/ocaml-lsp-server.1.20.0~5.3preview/opam [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  "Max Lantas <mnxndev@outlook.com>"
+]
+license: "ISC"
+homepage: "https://github.com/ocaml/ocaml-lsp"
+bug-reports: "https://github.com/ocaml/ocaml-lsp/issues"
+depends: [
+  "dune" {>= "3.0"}
+  "ocaml" {>= "5.3" & < "5.4"}
+  "yojson"
+  "base"
```
Comment (voodoos) [inline packages/ocaml-lsp-server/ocaml-lsp-server.1.20.0~5.3preview/opam:24]:
```suggestion
  "base" {>= "0.16.0"}
```

On my environment the build passes with 0.16.0

File: packages/dot-merlin-reader/dot-merlin-reader.5.3~5.3preview/opam [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+bug-reports:  "https://github.com/ocaml/merlin/issues"
+dev-repo:     "git+https://github.com/ocaml/merlin.git"
+license:      "MIT"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "5.3" & < "5.4"}
+  "dune" {>= "2.9.0"}
```
Comment (voodoos) [inline packages/dot-merlin-reader/dot-merlin-reader.5.3~5.3preview/opam:15]:
```suggestion
  "dune" {>= "3.0.0"}
```

File: packages/ocaml-lsp-server/ocaml-lsp-server.1.20.0~5.3preview/opam (hunk 3 above)
Comment (kit-ty-kate) [reply packages/ocaml-lsp-server/ocaml-lsp-server.1.20.0~5.3preview/opam:24]: significant v prefix
```suggestion
  "base" {>= "v0.16.0"}
```

**Comment by mseri** [comment]
Is this ready for merge?

**Comment by kit-ty-kate** [comment]
yes

**Pull request merged**